### PR TITLE
Finalize result index

### PR DESCRIPTION
**Pull request by dwalintukan**

### CODE REVIEW

File: src/schema/index.js [hunk 1]
```diff
@@ -185,6 +185,7 @@ type Mutation {
     senderAddress: String!
     topicAddress: String!
     oracleAddress: String!
+    optionIdx: Int!
```
Comment (frankobe) [inline src/schema/index.js]: as discussed, lets keep mutation as close to smart contract interface and grab the optionIdx from the last oracle

File: src/schema/resolvers.js [hunk 2]
```diff
@@ -577,6 +577,7 @@ module.exports = {
         version,
         topicAddress,
         oracleAddress,
+        optionIdx,
```
Comment (frankobe) [inline src/schema/resolvers.js]: removed

File: src/schema/index.js (hunk 1 above)
Comment (dwalintukan) [reply src/schema/index.js]: 👍

File: src/schema/resolvers.js (hunk 2 above)
Comment (dwalintukan) [reply src/schema/resolvers.js]: 👍

File: src/schema/resolvers.js [hunk 3]
```diff
@@ -593,16 +593,31 @@ module.exports = {
         throw err;
       }
 
+      // Fetch oracle to get the finalized result
+      const oracle = await Oracles.findOne({ address: oracleAddress }, { options: 1, optionIdxs: 1 });
+      let winningIndex;
+      if (!oracle) {
+        logger.error(`Could not find Oracle ${address} in DB.`);
```
Comment (frankobe) [inline src/schema/resolvers.js]: we should ensure oracle exists before call smart contract. If oracle not found, throw an error to FE

Comment (dwalintukan) [reply src/schema/resolvers.js]: done

File: src/schema/resolvers.js [hunk 4]
```diff
@@ -572,14 +572,29 @@ module.exports = {
...
       } = data;
 
+      // Fetch oracle to get the finalized result
+      const oracle = await Oracles.findOne({ address: oracleAddress }, { options: 1, optionIdxs: 1 });
+      let winningIndex;
+      if (!oracle) {
+        logger.error(`Could not find Oracle ${address} in DB.`);
+        throw new Error(`Could not find Oracle ${address} in DB.`);
+      } else {
+        for (let i = 0; i < oracle.options.length; i++) {
```
Comment (frankobe) [inline src/schema/resolvers.js]: nit: add a comment to explain comparison between `options` & `optionIdxs`

Comment (dwalintukan) [reply src/schema/resolvers.js]: done

**Pull request merged**